pkg/controller/atlasdeployment/searchindex.go

```diff
@@ -0,0 +1,178 @@
+package atlasdeployment
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/searchindex"
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1/status"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/workflow"
+)
+
+const (
+	IndexStatusActive = "STEADY"
+)
+
+type searchIndexReconciler struct {
+	ctx        *workflow.Context
+	deployment *akov2.AtlasDeployment
+	k8sClient  client.Client
+	projectID  string
+	indexName  string
+}
+
+func (sr *searchIndexReconciler) Reconcile(stateInAKO, stateInAtlas *searchindex.SearchIndex, errs []error) workflow.Result {
+	sr.ctx.Log.Debugf("starting reconciliation for index '%s'", sr.indexName)
+	defer sr.ctx.Log.Debugf("finished reconciliation for index '%s'", sr.indexName)
+
+	if len(errs) != 0 {
+		return sr.terminate(&searchindex.SearchIndex{
+			SearchIndex:                akov2.SearchIndex{Name: sr.indexName},
+			AtlasSearchIndexConfigSpec: akov2.AtlasSearchIndexConfigSpec{},
+			ID:                         nil,
+			Status:                     nil,
+		}, errors.Join(errs...))
+	}
+
+	emptyInAtlas := stateInAtlas == nil
+	emptyInAKO := stateInAKO == nil
+
+	var currentStatus string
+	if stateInAtlas != nil && stateInAtlas.Status != nil {
+		currentStatus = *stateInAtlas.Status
+	}
+	switch currentStatus {
+	case IndexStatusActive, "":
+		break
+	default:
+		return sr.progress(stateInAtlas)
+	}
```

Comment:
```suggestion
	if currentStatus != IndexStatusActive || currentStatus != "" {
		return sr.progress(stateInAtlas)
	}
```

Reply:
This condition is always try, no?

Reply:
Indeed! this should be:

```suggestion
	if currentStatus != IndexStatusActive {
		return sr.progress(stateInAtlas)
	}
```